css/amp.css

```diff
@@ -78,6 +78,15 @@ body {
   right: 0;
 }
 
+.-amp-layout-flex-item {
+  display: block;
+  position: relative;
+  flex: 1 1 0;
```

Comment:
tested on property 1 and 2, made the change. 
For 3, the width and height `100%` works as expected. 
If the parent is `display:flex` setting width&height to `100%` won't be necessary. 
If parent is NOT `display:flex` children will need a width/height. So do we want the width and height `100%` in css in case parent is NOT `display:flex`?